api/api.proto

```diff
@@ -803,6 +803,206 @@ service OpenStorageVolume {
     }
 }
 
+service OpenStorageCredentials {
+
+  // Provide credentials to OpenStorage and if valid,
+  // it will return an identifier to the credentials
+  rpc ProvideForAWS(ProvideCredentialsForAWSRequest)
+    returns (ProvideCredentialsForAWSResponse) {
+      option(google.api.http) = {
+        post: "/v1/credentials/provide/aws"
+        body: "*"
+      };
+    }
+
+  // Create and store credentials
+  // Change like ProvideForAWS
+  rpc ProvideForAzure(ProvideCredentialsForAzureRequest)
+    returns (ProvideCredentialsForAzureResponse) {
+      option(google.api.http) = {
+        post: "/v1/credentials/create/azure"
+        body: "*"
+      };
+    }
+
+  // Create and store credentials
+  // Change like ProvideForAWS
+  rpc ProvideForGoogle(ProvideCredentialsForGoogleRequest)
+    returns (ProvideCredentialsForGoogleResponse) {
+      option(google.api.http) = {
+        post: "/v1/credentials/create/google"
+        body: "*"
+      };
+    }
+  // EnumerateForAWS lists the configured AWS credentials                      
+   rpc EnumerateForAWS(EnumerateCredentialsForAWSRequest)                       
+     returns (EnumerateCredentialsForAWSResponse) {                             
+     option(google.api.http) = {                                                
+       post: "/v1/credentials/enumerate/aws"                                    
```

Comment:
These endpoints look wrong, it should look like the current endpoints and you shouldn't need a different type of endpoint for each type of cloud.
For creating POST to /v1/credential (The input object should have the type and corresponding config)
For listing GET from /v1/credential
For getting a particular credential GET from /v1/credential/{id}
For deleting DELETE from /v1/credential/{id}

Also preferably it should be "credential" and not "credentials"

Reply:
@disrani-px Do not confuse these with the REST endpoint. This is the REST Gateway for the gRPC. And all are POSTs since they are all pushing objects.

Reply:
I didn't say they were the same, but the endpoints should be similar. Also POST doesn't mean you are pushing objects, you can always send objects with other types of requests too. POST should be used for create operations and the other request types should be used as mentioned above for other operations.

Reply:
I think this discussion is outside the scope of this PR. We can take this offline. @ipochi For now continue with all POSTs as the others, until further notice. @disrani-px and I will discuss it offline and propose a PR change if needed.